src/tiled/mapeditor.cpp

```diff
@@ -404,6 +404,8 @@ void MapEditor::setCurrentDocument(Document *document)
     if (mapDocument) {
         connect(mapDocument, &MapDocument::currentLayerChanged,
                 this, &MapEditor::updateLayerComboIndex);
+        connect(mapDocument, &MapDocument::tileProbabilityChanged,
+                mStampBrush, &StampBrush::updateRandomList);
```

Comment:
It is really the responsibility of the stamp brush itself to update its random list when needed. The current map document is already available to the stamp brush, so it could make this connection itself (and keep `updateRandomList` private).

Also, please note what I wrote at #1996 about only "invalidating" the random picker when a tile probability changes rather than immediately updating it.